apps/scan/backend/src/scanners/custom/state_machine.ts

```diff
@@ -827,7 +834,21 @@ function buildMachine({
           on: {
             SCAN: 'scanning',
             SCANNER_NO_PAPER: 'no_paper',
-            SCANNER_READY_TO_SCAN: doNothing,
+            SCANNER_READY_TO_SCAN: 'check_ready_to_scan',
+          },
+        },
+        check_ready_to_scan: {
+          invoke: {
+            src: async () => isReadyToScan({ auth, store, usbDrive }),
+            onDone: [
+              {
+                target: 'scanning',
+                cond: (_context, event) => event.data,
+              },
+              {
+                target: 'ready_to_scan',
```

Comment:
The naming is a little confusing here - we transition back to `ready_to_scan` when `isReadyToScan` returns false. What do you think about delineating eg `hardware_ready_to_scan` and `appStateIsReadyToScan`?